ixmp4/data/db/iamc/datapoint/repository.py

```diff
@@ -106,8 +124,12 @@ def join_auth(self, exc: db.sql.Select) -> db.sql.Select:
 
         return exc
 
-    def select_joined_parameters(self, join_runs=False):
-        bundle = []
+    def select_joined_parameters(
+        self, join_runs: bool = False
+    ) -> db.sql.Select[tuple[DataPoint]]:
+        # NOTE Not quite sure about this bundle, seems to possibly take all types of
+        # all model classes?
+        bundle: list[db.Label[str] | db.Label[int] | db.Bundle[Any]] = []
```

Comment:
So `bundle` was a `list` already and I essentially just added the types that get added to it from the `extend()` calls below. `Model.name` would be a simple `str`, I think, but `Model.name.label("model")` returns a `Label[str]`, which is different for sqlalchemy: the `.label("model")` translates to `[SELECT ... FROM ...] AS model`. This is a different entity and thus requires the `Label` type.
I also don't think we can make a `Select[Bundle[...]]` work. Compare where this function is used: in `DataPointRepository.select()`, we craft an SQL statement as `exc`, which is either directly provided, coming from our function or created as `db.select(self.bundle)`. Since all these options are stored in the same variable, they should all have the same type. And `db.select(self.bundle)` simply returns a `Select[Tuple[Any]]` (where self.bundle is annotated generically as `Bundle[Any]`). Within `select_joined_parameters()`, we also don't return the `bundle`, but a `db.select(bundle)`, which turns into a `Select[Tuple[...]]`. 
Hope this makes sense and is fine :)